android/app/src/main/java/com/dismoi/scout/Floating/Layout/Bubble.java

```diff
@@ -0,0 +1,211 @@
+package com.dismoi.scout.Floating.Layout;
+
+import android.animation.AnimatorInflater;
+import android.animation.AnimatorSet;
+import android.content.Context;
+import android.graphics.Point;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.AttributeSet;
+import android.util.DisplayMetrics;
+import android.view.Display;
+import android.view.MotionEvent;
+import android.view.WindowManager;
+
+import com.dismoi.scout.R;
+
+public class Bubble extends Base {
+  private float initialTouchX;
+  private float initialTouchY;
+  private int initialX;
+  private int initialY;
+  private OnBubbleRemoveListener onBubbleRemoveListener;
+  private OnBubbleClickListener onBubbleClickListener;
+  private static final int TOUCH_TIME_THRESHOLD = 150;
+  private long lastTouchDown;
+  private final MoveAnimator animator;
+  private int width;
+  private final WindowManager windowManager;
+  private boolean shouldStickToWall = true;
+
+  public void setOnBubbleRemoveListener(OnBubbleRemoveListener listener) {
+    onBubbleRemoveListener = listener;
+  }
+
+  public void setOnBubbleClickListener(OnBubbleClickListener listener) {
+    onBubbleClickListener = listener;
+  }
+
+  public Bubble(Context context) {
+    super(context);
+
+    animator = new MoveAnimator();
```

Comment:
Could these be set in parent class?
Or could these assignations (in every custom constructor) be done in the same location to make it DRY?